doc/async_context.md

```diff
@@ -0,0 +1,149 @@
+# AsyncContext
+
+The [AsyncWorker](async_worker.md) class may not be appropriate for every scenario, because with
+those the async execution still happens on the main event loop. When using any
```

Comment:
I don't think this is quite right as the async execution happens in a thread from the existing thread pool. I'll have a closer look and add some I think describes the difference.

Reply:
I matched up with https://nodejs.org/api/n-api.html#n_api_custom_asynchronous_operations.